### PR TITLE
Fix _onModelEvent regression

### LINKED ISSUE #3662: .trigger('change')
v1.2.0 line 1117 doesn't contain 'model &&'
When setting value (using "silent:true") and then chaining & triggering a "change"  event, it throws an error

model.set({'some':'data'}, {'silent':true}).trigger('change');

I know it says "not" to use silent...

### LINKED ISSUE #3717: Manually triggering event on model that is part of collection causes error
Hello -- we were upgrading backbone in our app and ran into an unexpected error.  Here is the simplest repro for it:
```
var c = new Backbone.Collection();
var m = new Backbone.Model();

c.add(m);
m.trigger('change');  //  error thrown
```
Was there an intentional change in how events are handled?

Thanks!

### PR DESCRIPTION
**Pull request by jridgewell**
Fixes #3662. Fixes #3717

### CODE REVIEW

**Comment by megawac** [comment]
Could you add a test

File: backbone.js [hunk 1]
```diff
@@ -1148,14 +1148,16 @@
...
     _onModelEvent: function(event, model, collection, options) {
-      if ((event === 'add' || event === 'remove') && collection !== this) return;
-      if (event === 'destroy') this.remove(model, options);
-      if (event === 'change') {
-        var prevId = this.modelId(model.previousAttributes());
-        var id = this.modelId(model.attributes);
-        if (prevId !== id) {
-          if (prevId != null) delete this._byId[prevId];
-          if (id != null) this._byId[id] = model;
+      if (model) {
```
Comment (megawac) [inline backbone.js]: should we do `model instanceof Backbone.Model` instead?

Comment (jridgewell) [inline backbone.js]: I don't think it'll be necessary. The issue was people triggering `change` without the required parameters (just `trigger('change')`).

**Comment by jridgewell** [comment]
Test added.

**Comment by QuotableWater7** [comment]
thanks guys! :+1:

**Comment by QuotableWater7** [comment]
I just looked at the code - does this change mean that triggering `change` manually would fail silently?

**Comment by QuotableWater7** [comment]
ah my bad, the change makes sense to me now - thanks again.

**Pull request merged**